### ISSUE TITLE
Incomplete encoding of break

### ISSUE THREAD

**Issue by bobismijnnaam** [marcoeilers/nagini#141]
I ran into a strange issue yesterday. It concerns the following code snippet:

```
from nagini_contracts.contracts import *

class MyClass:
    def __init__(self) -> None:
        self.x = 0

def f(m: MyClass) -> None:
    Requires(Acc(m.x) and m.x == 0)
    Ensures(Acc(m.x))
    Ensures(m.x == 1)
    while True:
        Invariant(Acc(m.x) and m.x == 0)
        try:
            break
        finally:
            m.x = 1
```

This code should verify, but it currently does not:
```
$ nagini finally_after_break.py
Verification failed
Errors:
Postcondition of f might not hold. Assertion (m.x == 1) might not hold. (finally_after_break.py@10.12)
Verification took 8.56 seconds.
```

I think the problem is that currently break is encoded as a direct goto to after the nearest while loop, so the finally is skipped altogether. I think this can be solved by failing if this usage of break is detected, or handle breaks the same way returns are handled, i.e. with an extra auxiliary control flow code variable. In the second case the way finally is encoded will also need to be adjusted.

**Comment by marcoeilers** [comment]
Good find, that looks very unsound actually. I'll have a look at it later.

**Comment by marcoeilers** [comment]
Fixed in commit aef871aab24e81528a1de35409a1d913851ac280.
